src/pages/transfer.tsx

```diff
@@ -93,10 +95,34 @@ const TransferPage = () => {
         setAssets(_assets);
       }
     } else {
-      const _assets = await AssetRegistry.getAssetsOnBlockchain(
-        RELAY_CHAIN,
-        chains[sourceChainId].paraId
+      const chainData = (await getChains()).find(
+        (chain) => chain.paraId ?
+          chain.paraId === sourceChainId && chain.relay.id === RELAY_CHAIN
+          :
+          sourceChainId === 0 && chain.id === RELAY_CHAIN
       );
+
+      const _assets = [];
+      if (chainData) {
+        const tokens = (await getTokens()).filter((token) => {
+          const isPartOfSourceChain = token.data.id.startsWith(chainData.id);
+          return isPartOfSourceChain;
+        });
+        const assets: Asset[] = tokens.map(t => {
+          const asset: Asset = {
+            asset: "",
+            name: t.data.symbol,
+            symbol: t.data.symbol,
+            decimals: t.data.decimals,
+            xcmInteriorKey: t.data,
```

Comment:
is this correct?
`xcmInteriorKey: t.data`

Reply:
I did do this on purpose, however, you are absolutely right. This looks very hacky. I will do some renaming to make the code more readable.

Reply:
What I'm concerned about is if it is a right decision to use `t.data` as the xcmInteriorKey

Reply:
Oh, yeah it isn't actually an `xcmInteriorKey` in case of a non-cross-chain transfer. And in case of a cross-chain transfer, it will actually be an `xcmInteriorKey`. I will do some cleanup and refactoring to separate these two distinct things. This was just a way for me to quickly be able to write and test the code, but I forgot to clean it up after testing.

Reply:
Should be good now after my latest commit.